starknet-testnet/generateMarkdown.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+import sys
+
+from pathlib import Path
+from starkware.starknet.business_logic.internal_transaction_interface import (
+    TransactionExecutionInfo,
+)
+
+WARP_ROOT = Path(__file__).parents[1]
+TMP = WARP_ROOT / "benchmark" / "json"
+FILE_NAME = "data"
+
+
+def steps_in_function_deploy(contract_name: str, result: TransactionExecutionInfo):
+    json_path = os.path.abspath(TMP / (FILE_NAME + ".json"))
+    if os.path.exists(json_path):
+        with open(json_path, "r") as json_file:
+            benchmark_data = json.load(json_file)
+    else:
+        benchmark_data = {}
+
+    benchmark_data.setdefault(contract_name, {})[
+        "steps"
+    ] = result.call_info.cairo_usage.n_steps
+
+    with open(json_path, "w") as json_file:
+        json.dump(benchmark_data, json_file, indent=3)
+
+
+def steps_in_function_invoke(contract_name: str, result: TransactionExecutionInfo):
+    json_path = os.path.abspath(TMP / (FILE_NAME + ".json"))
+    if os.path.exists(json_path):
+        with open(json_path, "r") as json_file:
+            benchmark_data = json.load(json_file)
+    else:
+        benchmark_data = {}
+
+    benchmark_data.setdefault("_".join(contract_name.split("_")[:-1]), {})[
```

Comment:
What's the reasoning behind the name processing here?

Reply:
The names that are passed into `generateMarkdown.py` from the starknet testnet invoke command are of this format `unreachableCode_0d583609`
The above snippet was to keep everything as is but for the suffix. So that the names passed into the JSON/markdown are of the format `unreachableCode`